subprojects/model-core/src/main/java/org/gradle/api/internal/tasks/NodeExecutionContext.java

```diff
@@ -25,4 +25,6 @@ public interface NodeExecutionContext {
      * Locates the given execution service.
      */
     <T> T getService(Class<T> type) throws ServiceLookupException;
+
+    boolean isGlobal();
```

Comment:
💭 I am not very happy to extend the `NodeExecutionContext` by another method to denote whether execution happens as part of running an execution plan or outside of it, though I didn’t come up with a better solution.

Reply:
I'm not sure about "isGlobal()" as the method name, though, and maybe add some javadocs to explain what its contract is.

Also, I think we have a similar problem to the one we're trying to fix, where we have nodes that are scheduled but are executed by something else before the scheduler starts the node. In this case, the build operation will not be fired but the console will be expecting a build operation. Ideally `isGlobal()` would return true if the node had been scheduled, regardless of how its execution was triggered. However, this is really a separate issue, as in this case the progress would not get to 100%, rather than go past 100%.

Maybe it would be better for the execution plan to inform a node when it is a work graph, rather than have the context expose this. The transform nodes could potentially override `Node.prepareForScheduling()` for this.